src/ILCompiler/src/Program.cs

```diff
@@ -342,6 +353,7 @@ private int Run(string[] args)
                         if (entrypointModule != null)
                             throw new Exception("Multiple EXE modules");
                         entrypointModule = module;
+                        r2rCompilerContext?.InitializeAlgorithm(targetDetails, module.MetadataReader.GetTableRowCount(TableIndex.TypeDef));
```

Comment:
Hmm, I probably don't need to pass targetDetails around again, will look into fixing this.